### DISCUSSION TITLE
Is there any tutorial for aspnet core code base?

### DISCUSSION THREAD

**Discussion by abbasimo** [dotnet/aspnetcore#48729] [Q&A] [answered]
Hi there!
I want to figure out how framework works under hood and handle request in parallel. But I couldn't understand the code. Is there any tutorial about code base understanding? Something like that a person explains modules or anything that make a code base more clear for beginners?
Or do you have any suggestions that gave me a clue to start?
Thanks 🌹

**Comment by gfoidl** [comment, marked as answer]
>  Or do you have any suggestions that gave me a clue to start?

Personally I find it the easiest way by using the (Visual Studio) debugger. Set a breakpoint in your code, examine the stack-trace and then you can set breakpoints in the framework code too.
That way I learned how the pieces play together and got a rough understanding on how the things work. Based on that you can dig into the code deeper (and deeper, ...).

**Reply by abbasimo** [reply]
Thank you for your good suggestion and make my wrong issue to the discussion topic :)

**Reply by abbasimo** [reply]
I'm sorry for my stupid question. I cloned the repository and built it. but I don't know how execute it as a program and breakpoint.
I want have a Web API, and see what happens in the framework when I trigger a Web API. I don't know how make a Web API project in this case and what should be my first breakpoint. Can you make a example for me?

**Reply by gfoidl** [reply]
For that investigation you don't need to build the repository.

1. Open Visual Studio and create a new "Web API project"
2. in the code of the default sample just put a breakpoint in the api-method
3. in Visual Studio's option under Debugger enable "source link support"
4. run your created project under the debugger
5. execute the API, so that the breakpoint is hit
6. now you can start examining the code
